hibernate-core/src/main/java/org/hibernate/metamodel/internal/MetamodelImpl.java

```diff
@@ -625,7 +625,7 @@ else if ( result == INVALID_IMPORT ) {
 	 */
 	public String[] getImplementors(String className) throws MappingException {
 
-		if (implementorsCache.containsKey(className)) {
+		if (implementorsCache.get(className) != null) {
```

Comment:
Here you're still doing 2 map accesses. Sorry, I should have been more clear: you have to store the first one in a variable then reuse the variable for the test and the return.